test/mdn_test.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+import pytest
+import torch
+from torch import Tensor, eye, ones, zeros
+import torch.nn as nn
+from torch.distributions import MultivariateNormal
+
+from pyknos.mdn.mdn import MultivariateGaussianMDN
+
+
+def linear_gaussian(
+    theta: Tensor,
+    likelihood_shift: Tensor,
+    likelihood_cov: Tensor,
+) -> Tensor:
+
+    chol_factor = torch.cholesky(likelihood_cov)
+
+    return likelihood_shift + theta + torch.mm(chol_factor, torch.randn_like(theta).T).T
+
+
+@pytest.mark.parametrize("dim", ([1, 5, 10]),)
+def test_mdn_for_diff_dimension_data(dim: int, 
+            hidden_features: int=50, 
+            num_components: int=10) -> None:
+
+    theta = torch.rand(3, dim)
```

Comment:
I am not sure why we are using here theta instead of `x`. The tests should make sense in a general conditional-density estimation setting, and not only in a sbi context.